src/fetcher/mod.rs

```diff
@@ -0,0 +1,50 @@
+use super::{
+    interface::{ChainlinkContract, PriceData, Serializable},
+    Configuration, CryptoPrices,
+};
+
+/// Retrieves the price of an underlying asset from a particular contract
+async fn fetch_price_for_contract(
+    rustlink_configuration: Configuration,
+    address: &str,
+) -> Result<PriceData, alloy::contract::Error> {
+    let contract = ChainlinkContract::new(rustlink_configuration.provider, address).await?;
+    contract.get_price().await
+}
+
+/// A task that continously fetches cryptocurrency prices from Chainlink decentralised price data feed.
+pub async fn fetch_prices(crypto_prices: CryptoPrices) {
+    let contracts = &crypto_prices.configuration.contracts;
+    loop {
+        for contract_configuration in contracts {
+            let symbol = &contract_configuration.0;
+            let address = &contract_configuration.1;
+            match fetch_price_for_contract(crypto_prices.clone().configuration, address).await {
+                Ok(price_data) => {
+                    match crypto_prices
+                        .tree
+                        .insert(symbol, price_data.to_bin().unwrap())
+                    {
```

Comment:
I think you could perhaps use more idiomatic rust here:

```rust
fetch_price_for_contract(crypto_prices.clone().configuration, address)
  .await
  .and_then(|data| crypto_prices.tree.insert(...))
  .or_else(|err| err);
```

... or along these lines ...